python/ray/tests/conftest.py

```diff
@@ -223,9 +264,27 @@ def start_redis(db_dir):
         return address_str, processes
 
 
+def kill_all_redis_server():
+    import psutil
+
+    # Find Redis server processes
+    redis_procs = []
+    for proc in psutil.process_iter(["name", "cmdline"]):
+        if (
+            proc.info["name"] == "redis-server"
+            and "redis-server" in proc.info["cmdline"]
+        ):
+            redis_procs.append(proc)
+
+    # Kill Redis server processes
+    for proc in redis_procs:
+        proc.kill()
+
+
 @contextmanager
 def _setup_redis(request):
     with tempfile.TemporaryDirectory() as tmpdirname:
+        kill_all_redis_server()
```

Comment:
minor, but may be it could be after yield? (so we kill it as a part of current conftest, not killing redis from the previous conftest)

Reply:
I think maybe we should do both? The reason I added it here is because maybe the program crashed and the kill has never been reached